core/src/com/po/fuck/model/weapons/Glock.java

```diff
@@ -0,0 +1,25 @@
+package com.po.fuck.model.weapons;
+
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.graphics.g2d.Sprite;
+import com.badlogic.gdx.math.Vector2;
+import com.po.fuck.model.Entity;
+
+public final class Glock extends Gun {
+    {
+        // width = new Sprite(new Texture("glock3.png")).getWidth();
+        // height = new Sprite(new Texture("glock3.png")).getHeight();
+        cooldown = 0.5f;
+    }
+
+    public Glock(Entity owner) {
+        super(owner);
+    }
+
+    @Override
+    protected Bullet shoot(Vector2 muzzle_position, Vector2 direction) {
+        return new CosmicBullet(muzzle_position, new Sprite(new Texture("bullet2.png")).getWidth(),
```

Comment:
We should create some static factory(probably something like `CosmicBullet.create(position)`) that will only accept the position.

Reply:
What for? What is wrong with current implementation?

Reply:
We want to create a specific `CosmicBullet`, why are we passing it's texture's data, this class should know it already ?

Reply:
I am not sure that it is optimal, as we should still pass the direction to the bullet.  Thus, it would look like `CosmicBullet.create(position,direction)` which does not differ from the current constructor (after removing `height` and `width`).